libc-bottom-half/headers/public/__header_dirent.h

```diff
@@ -21,15 +21,15 @@
 #define IFTODT(x) (__wasilibc_iftodt(x))
 #define DTTOIF(x) (__wasilibc_dttoif(x))
 
-int __wasilibc_iftodt(int x);
-int __wasilibc_dttoif(int x);
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #include <__struct_dirent.h>
 #include <__typedef_DIR.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+int __wasilibc_iftodt(int x);
+int __wasilibc_dttoif(int x);
```

Comment:
It's a little untidy to include other headers inside the `extern "C"` block; could you move those `#include`s outside of the block?